AmplifyPlugins/DataStore/Tests/DataStoreHostApp/AWSDataStorePluginLazyLoadTests/LL5/Team1+Schema.swift

```diff
@@ -28,7 +28,7 @@ extension Team1 {
     model.fields(
       .field(team1.teamId, is: .required, ofType: .string),
       .field(team1.name, is: .required, ofType: .string),
-      .belongsTo(team1.project, is: .optional, ofType: Project1.self, targetNames: ["team1ProjectProjectId", "team1ProjectName"]),
+      .belongsTo(team1.project, is: .optional, ofType: Project1.self, associatedFields: [Project1.keys.projectId, Project1.keys.name], targetNames: ["team1ProjectProjectId", "team1ProjectName"]),
```

Comment:
These files, which are code generated from `amplify codegen models` process, need to be updated to generate `associatedFields`. We can definitely explore doing that in this file: https://github.com/aws-amplify/amplify-codegen/pull/541/files#diff-eaba881c8c73aea316d5f63356d638f20a3bf275c7a4b66d8e97e1070c2de177 similar to how we've added the `associatedFields` for has-many and has-one model fields which replaces "associatedWith", but I'm evaluating whether we really need to, or if we already have all the information. 

## Codegen

<img width="1090" alt="image" src="https://user-images.githubusercontent.com/1365977/217274766-3b1ad486-f2a6-4a79-838d-96527a623b0b.png">

In the Codegen code, the swift visitor doesn't populate `connectionInfo` and is processed at the platform agnostic level. There will be a much heavier lift to achieve this by populating connectionInfo with associatedFields and then codegenerating the info into the swift schema model files.

## has-many

has-many "associatedFields" is required because we're storing the parent's foreign key in the lazy child list. We don't know which FK on the child is the parent for uni-directional has-many unless we pass that information to the child lazy object from the parent. We were already generating "associatedWith" but should have been generating `associatedFields` for CPK uni-directional has-many use case. Lazy loading a list of child models uses the parent's FK fields on the child and the predicate values are the parent's identifiers, ie. `query(Child.self, where: Child.keys.parentFK == "parentId123")` 

## has-one

Evaluating whether the library needs to use the information of the has-one "associatedFields" / "associatedWith" for our use cases. It uses targetNames to extract out the value, ie. `element[targetname]`. Then the primary key values should be passed to the LazyReference.

LazyReference child only needs its own primary key (the child primary key values). DataStore can lazy load itself by using `ModelType.schema.primaryKey.sqlName` as the key since it will resolve to "id" for non-CPK and `@@primaryKey` for CPK use cases. If `@@primaryKey` then concat the identifiers using "#" to get the final query to be `query(Child.self, where: "@@primaryKey" == "id123#title123".  The child's schema already knows which fields are the primary keys, we only need the value (or values). 

Syncing a parent only needs its the child primary key values as well. API Model helpers can translate the parent object to the GraphQL input by pulling out the child primary key values from the not loaded Lazy Reference and then use the targetName as the key:
```
 { 
"projectid": "project123",
...
"hasOneParentChildId": "childId123" // "childId123" pulled from the LazyReference, targetName "hasOneParentChildId" pulled the parent's hasOne modelField's targetNames
}
```

It doesn't seems like we need `associatedWith`/`associatedFields` at all. The parent element and the targetNames gives us the child primary key values, the child schema gives us the primary key field names.

## belongs-to

Evaluating whether the library needs to use the belongs-to's "associatedFields" for our use cases. Applying the same logic from the has-one case to see if it checks out:

Use the targetNames to extract out the value, ie. `ChildElement[targetname]`. Then the primary key values should be passed to the LazyReference parent.

LazyReference parent only needs its own primary key (the parent's primary key values). DataStore can lazy load itself by using `ModelType.schema.primaryKey.sqlName` as the key since it will resolve to "id" for non-CPK and `@@primaryKey` for CPK use cases. If `@@primaryKey` then concat the identifiers using "#" to get the final query to be `query(Parent.self, where: "@@primaryKey" == "id123#title123".  The child's schema already knows which fields are the primary keys, we only need the value (or values). 

Syncing a child only needs its the parent primary key values as well. API Model helpers can translate the child object to the GraphQL input by pulling out the parent primary key values from the not loaded Lazy Reference and then use the targetName as the key:
```
 { 
"id": "parent123",
...
"childParentId": "parentId123" // "parent123" pulled from the LazyReference, targetName "chidlParentId" pulled the child's belongsTo modelField's targetNames
}
```